src/commands.ts

```diff
@@ -0,0 +1,174 @@
+import { FlatpakManifest, Module } from './flatpak.types'
+import { Command } from './terminal'
+import { getuid } from 'process'
+
+
+export const getBuildAppCommand = (
+  module: Module,
+  cwd: string,
+  buildDir: string,
+  buildArgs: string[]
+): [Command[], Command[]] => {
+  let buildAppCommand: Command[] = []
+  let rebuildAppCommand: Command[] = []
+  const configOpts = (module['config-opts'] || []).join(' ')
+
+  switch (module.buildsystem) {
+    case 'meson':
+      {
+        const mesonBuildDir = '_build'
+        buildArgs.push(`--filesystem=${cwd}/${mesonBuildDir}`)
+        rebuildAppCommand = [
+          new Command('flatpak', [
+            'build',
+            ...buildArgs,
+            buildDir,
+            'ninja',
+            '-C',
+            mesonBuildDir,
+          ],cwd),
+          new Command('flatpak', [
+            'build',
+            ...buildArgs,
+            buildDir,
+            'meson',
+            'install',
+            '-C',
+            mesonBuildDir,
+          ],cwd),
+        ]
+
+        buildAppCommand = [
+          new Command('flatpak', [
+            'build',
+            ...buildArgs,
+            buildDir,
+            'meson',
+            '--prefix',
+            '/app',
+            mesonBuildDir,
+            configOpts,
+          ],cwd),
+          ...rebuildAppCommand,
+        ]
+      }
+      break
+    case 'simple':
+      {
+        const buildCommands = module['build-commands'].map((command) => {
+          return new Command('flatpak', [
+            'build',
+            ...buildArgs,
+            buildDir,
+            command,
+          ],cwd)
+        })
+        rebuildAppCommand = buildCommands
+        buildAppCommand = buildCommands
+      }
+      break
+  }
+  return [buildAppCommand, rebuildAppCommand]
+}
+
+export const exportBundle = (): Command => {
+  return new Command('flatpak-builder', [])
+}
+
+export const buildDependencies = (manifestPath: string, buildDir: string, cwd: string, stateDir?: string, stopAt?: string): Command => {
+  const args = [
+    '--ccache',
+    '--force-clean',
+    '--disable-updates',
+    '--disable-download',
+    '--build-only',
+    '--keep-build-dirs',
+  ]
+  if (stateDir) {
+    args.push(`--state-dir=${stateDir}`)
+  }
+  if (stopAt) {
+    args.push(`--stop-at=${stopAt}`)
+  }
+  args.push(buildDir)
+  args.push(manifestPath)
+
+  return new Command('flatpak-builder', args, cwd)
+}
+
+export const updateDependencies = (manifestPath: string, buildDir: string, cwd: string, stateDir?: string, stopAt?: string): Command => {
```

Comment:
Yeah, indeed updateDependencies and buildDependencies are the same.
I assume this is a copy-paste error :)

Reply:
Fixed with the latest commit